satellite/notifications/service.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package notifications
+
+import (
+	"context"
+	"sync"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/rpc"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/overlay"
+)
+
+var (
+	mon = monkit.Package()
+
+	Error = errs.Class("notification service error")
+)
+
+type Service struct {
+	log     *zap.Logger
+	dialer  rpc.Dialer
+	overlay *overlay.Service
+	sending sync.WaitGroup
+}
+
+func NewService(log *zap.Logger, dialer rpc.Dialer, overlay *overlay.Service) *Service {
+	return &Service{
+		log:     log,
+		dialer:  dialer,
+		overlay: overlay,
+		sending: sync.WaitGroup{},
+	}
+}
+
+func (service *Service) Notify(ctx context.Context, nodeID storj.NodeID, notification *pb.Notification) {
+	defer mon.Task()(&ctx)(nil)
+
+	// TODO: add limits
+	service.sending.Add(1)
```

Comment:
This could end up racing with close.

Reply:
what did I do wring? I think that is the same as we have in `mailservice`

Reply:
This `Add` might end up being called after `Wait` returns.

Reply:
do you mean that I need to check some atomic var before, to check if service has been closed?